npep/npep-133.md

```diff
@@ -0,0 +1,140 @@
+# NPEP-133: FQDN Selector for Egress Traffic
+
+* Issue:
+  [#133](https://github.com/kubernetes-sigs/network-policy-api/issues/133)
+* Status: Provisional
+
+## TLDR
+
+This enhancement proposes adding a new optional selector to specify egress peers
+using [Fully Qualified Domain Names](https://www.wikipedia.org/wiki/Fully_qualified_domain_name)
+(FQDNs).
+
+## Goals
+
+* Provide a selector to specify egress peers using a Fully Qualified Domain Name
+  (for example `kubernetes.io`).
+* Support a restricted set of regex matching capabilities when specifying FQDNs.
+* Currently only `ALLOW` type rules are proposed.
+  * Correctly enforcing DENY rules based on FQDN selectors is difficult as there
+    is no guarantee a Network Policy plugin is aware of all IPs backing a FQDN
+    policy. 
+* Currently only AdminNetworkPolicy is the intended scope for this proposal.
+  * Since Kubernetes NetworkPolicy does not have a FQDN selector, adding this
+    capability to BaselineAdminNetworkPolicy can result in unintended behavior.
+    For example, if BANP allows traffic to `example.io`, but the namespace admin
+    installs a Kubernetes Network Policy, the namespace admin has no way to
+    replicate the `example.io` selector using just Kubernetes Network Policies.
+
+## Non-Goals
+
+* This enhancement does not include a FQDN selector for allowing ingress
+  traffic.
+* This enhancement does not include any L7 matching or filtering capabilities,
+  like matching HTTP traffic or URL paths.
+  * This selector should not control what DNS records are resolvable from a
+    particular workload.
+* This enhancement does not provide a mechanism for selecting in-cluster
+  endpoints using FQDNs. This is explicitly disallowed by the spec.
+  * To select Pods, Nodes, or the API Server, AdminNetworkPolicy has other more
+    specific selectors.
```

Comment:
Ack -- I reworded to be more inline with our discussion for Egress CIDRs.

I think leaving it undefined is the right course of action. While for Pods it's pretty trivial to expect providers to do the expected thing (it's the same as putting the Pod's IP directly in). But for something like Services things get tricky since often times providers have already load balanced Service VIPs to a Pod IP before NetPol enforcement.

We can provided better guidance and UX by using more specific selectors for K8s endpoints, so leaving this undefined seems like the best compromise. LMK what you think.